zebra-chain/src/sapling/shielded_data.rs

```diff
@@ -42,7 +65,10 @@ pub struct ShieldedData {
     pub binding_sig: Signature<Binding>,
 }
 
-impl ShieldedData {
+impl<T> ShieldedData<T>
+where
+    T: AnchorVariant,
+{
     /// Iterate over the [`Spend`]s for this transaction.
     pub fn spends(&self) -> impl Iterator<Item = &Spend> {
```

Comment:
You might want to return `Spend<PerSpendAnchor>` here.

For `ShieldedData<SharedAnchor>`, copy the shared anchor into each `Spend<PerSpendAnchor>`.